### PR TITLE
Timeseries csv parser

### PR DESCRIPTION
**Pull request by marcosmc**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
In the current implementation, TimeSeries only imports CSV files if the data is represented in a regular time series and the file has a version column.

**What is the new behavior (if this is a feature change)?**
The import of CSV files has been modified to make the version column optional, as well as to include irregular series and the possibility of defining the time column with long values.

### CODE REVIEW

**Review by zamarrenolm — APPROVED** [review]

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (geofjamg) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:157]: This is too much parameters, add a `TimeSeries.CsvParserConfig`.

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (geofjamg) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:545]: static final is useless

**Review by geofjamg — CHANGES_REQUESTED** [review]
It is fine to be able to configure CSV parsing to support other use case, but to be consistent we should also allow writing CSV with same config to keep things symmetrical. Here you only updated reading part.

**Comment by marcosmc** [comment]
> It is fine to be able to configure CSV parsing to support other use case, but to be consistent we should also allow writing CSV with same config to keep things symmetrical. Here you only updated reading part.

We have another PR waiting for the current one to be merged with the modifications to write CSV with the same configuration options. We have decided to separate it into 2 PR so that these are reduced in size and therefore faster to review.

**Comment by geofjamg** [comment]
> 
> 
> > It is fine to be able to configure CSV parsing to support other use case, but to be consistent we should also allow writing CSV with same config to keep things symmetrical. Here you only updated reading part.
> 
> We have another PR waiting for the current one to be merged with the modifications to write CSV with the same configuration options. We have decided to separate it into 2 PR so that these are reduced in size and therefore faster to review.

This is a bad practice, this is a small PR and everything should be consistent after the merge.

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: Done

**Comment by marcosmc** [comment]
CSV writer joined the current PR.

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:204]: Same comment there: you could also fix the declaration of the class and the fields

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:425]: I wonder if the separator should be added in the CsvParserConfig and then pass the config to this method

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:403]: I wonder if the separator should be added in the config and the pass only the config to this method

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:361]: Is an empty timeseries totally dumb? Maybe it could be useful for the timeline (see my comment about null TS vs empty TS in #1539)

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:299]: Add a default case that throws an AssertionError with the unexpected timeFormat

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:371]: What did you have in mind by replacing the equality by this epsilon? Maybe you should also introduce a constant instead of a magic number. What is the unit of this value?

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java:557]: I wonder if the flush should be moved in the writeCsv() method to avoid user to forget it. What do you think?

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java:696]: Add a default case that throws an exception

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java:692]: Is this code equivalent to:
```
writer.write(Double.toString(time / 1000.0));
```

I think that `time / 1000.0` will be automatically cast into a double (thanks to the `.0`)

**Review by mathbagu — CHANGES_REQUESTED** [review]

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: To avoid duplicate classes, an individual class is generated to group it with the CsvConfig class of TimeSeriesTable.

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:275]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java]: Replaced with an equality

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java:662]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java
Comment (marcosmc) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java:337]: At this point it only affects the reading of TimeSeries from a CSV file, it is possible to create an empty object of type TimeSeries without using a CSV file as a starting point

**Review by geofjamg — APPROVED** [review]

**Pull request merged**
